### PR TITLE
Implement sidemenu on frontend

### PR DESCRIPTION
**Pull request by ssbenedek**
### What is the purpose of this PR?

- Created the sidemenu component
- Separated the buttons into .tsx files

### How did you verify your code is stable and works correctly?

- Tested on my local machine

### CODE REVIEW

File: components/Sidemenu.tsx [hunk 1]
```diff
@@ -1,6 +1,41 @@
+'use client'
+
+// COMPONENTS
+import Link from "next/link";
```
Comment (kamka427) [inline components/Sidemenu.tsx:4]:
```suggestion

```

File: components/SidemenuButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import AnalyticsIcon from '@mui/icons-material/Analytics';
+
+// COMPONENTS
+import Link from "next/link";
+
+type SidemenuButtonProps = {
+    button_type: string;
+};
+
+const SidemenuButton: React.FC<SidemenuButtonProps> = ({ button_type }) => {
```
Comment (kamka427) [inline components/SidemenuButton.tsx:12]:
```suggestion
const SidemenuButton: React.FC<SidemenuButtonProps> = ({ button_type }: ChatHistoryButtonProps) => {
```

File: components/ChatHistoryButton.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// COMPONENTS
+import Link from "next/link";
+
+type ChatHistoryButtonProps = {
+    chat_id: number;
+};
+
+const ChatHistoryButton: React.FC<ChatHistoryButtonProps> = ({ chat_id }) => {
```
Comment (kamka427) [inline components/ChatHistoryButton.tsx:8]:
```suggestion
const ChatHistoryButton: React.FC<ChatHistoryButtonProps> = ({ chat_id }: ChatHistoryButtonProps) => {
```

**Review by kamka427 — COMMENTED** [review]
Looks great, great work. Please check the linter issues, I commented a few tips for getting around the warnings. Please don't forget to run Prettier after the changes with ```npm run format```.

**Review by kamka427 — APPROVED** [review]

**Review by MilosAlex — APPROVED** [review]
LGTM

**Pull request merged**